src/Core/src/Platform/iOS/BorderCALayer.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using CoreAnimation;
+using CoreGraphics;
+using Foundation;
+using UIKit;
+
+namespace Microsoft.Maui
+{
+    public class BorderCALayer : CALayer
+    {
+        CGRect _bounds;
+
+        CornerRadius _cornerRadius;
+
+        UIColor? _backgroundColor;
+
+        float _borderWidth;
+        UIColor? _borderColor;
+
+        public BorderCALayer()
+        {
+            _bounds = new CGRect();
+
+            ContentsScale = UIScreen.MainScreen.Scale;
+        }
+
+        public override void LayoutSublayers()
+        {
+            base.LayoutSublayers();
+
+            if (Bounds.Equals(_bounds))
+                return;
+
+            _bounds = new CGRect(Bounds.Location, Bounds.Size);
+        }
+
+        public override void DrawInContext(CGContext ctx)
+        {
+            base.DrawInContext(ctx);
+
+            ctx.AddPath(GetClipPath());
+            ctx.Clip();
+
+            DrawBackground(ctx);
+            DrawBorder(ctx);
+        }
+
+        public void UpdateCornerRadius(CornerRadius cornerRadius)
+        {
+            _cornerRadius = cornerRadius;
+
+            SetNeedsDisplay();
+        }
+
+        public void UpdateBackgroundColor(Color backgroundColor)
+        {
+            _backgroundColor = backgroundColor.ToNative();
+
+            SetNeedsDisplay();
+        }
+
+        public void UpdateBorderColor(Color borderColor)
+        {
+            _borderColor = borderColor.ToNative();
+
+            SetNeedsDisplay();
+        }
+
+        public void UpdateBorderWidth(double borderWidth)
+        {
+            _borderWidth = (float)borderWidth;
+
+            SetNeedsDisplay();
+        }
+
+        CGPath? GetClipPath() =>
+            GetRoundCornersPath(Bounds, _cornerRadius, _borderWidth).CGPath;
+
+        void DrawBackground(CGContext ctx)
+        {
+            if (_backgroundColor != null)
+            {
+                ctx.SetFillColor(_backgroundColor.CGColor);
+                ctx.AddPath(GetClipPath());
+                ctx.DrawPath(CGPathDrawingMode.Fill);
+            }
+        }
+
+        void DrawBorder(CGContext ctx)
+        {
+            if (_borderWidth == 0)
+                return;
+
+            ctx.SetLineWidth(2 * _borderWidth);
+            ctx.AddPath(GetClipPath());
+
+            if (_borderColor != null)
+            {
+                ctx.SetStrokeColor(_borderColor.CGColor);
+                ctx.DrawPath(CGPathDrawingMode.Stroke);
+            }
+        }
+
+        UIBezierPath GetRoundCornersPath(CGRect bounds, CornerRadius cornerRadius, float borderWidth = 0f)
```

Comment:
This will be removed and will use Microsoft.Maui.Graphics to get the path with corner radius.